### PR TITLE
Remove clusterid tag

### PR DESCRIPTION
**Pull request by spangenberg**
/cc @enxebre

### CODE REVIEW

File: pkg/actuators/machine/utils.go [hunk 1]
```diff
@@ -81,8 +82,8 @@ func getInstances(machine *machinev1.Machine, client awsclient.Client, instanceS
 			Values: []*string{&machineName},
 		},
 		{
-			Name:   aws.String("tag:clusterid"),
-			Values: []*string{&clusterID},
+			Name:   aws.String("tag:kubernetes.io/cluster/" + clusterID),
```
Comment (enxebre) [inline pkg/actuators/machine/utils.go]: can we move `kubernetes.io/cluster/` to a constant? can we also use use `sprintf` here?

File: pkg/actuators/machine/utils.go [hunk 2]
```diff
@@ -81,8 +82,8 @@ func getInstances(machine *machinev1.Machine, client awsclient.Client, instanceS
...
+			Values: []*string{&owned},
 		},
```
Comment (enxebre) [inline pkg/actuators/machine/utils.go]: can we put owned into a constant e.g `ResourceLifecycleOwned`? can we also use `aws.StringSlice` similar to line 262?

Comment (spangenberg) [reply pkg/actuators/machine/utils.go]: I tried, was not overly happy about the pointer from the const but I'll try to get it working.

File: pkg/actuators/machine/utils.go (hunk 1 above)
Comment (spangenberg) [reply pkg/actuators/machine/utils.go]: Sure, will make it a const.

Comment (enxebre) [reply pkg/actuators/machine/utils.go]: we could even have a function e.g `clusterKey/clusterIDTag` that returns  `fmt.Sprintf("%s%s", constant, name) see https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/pkg/cloud/aws/tags/cluster.go#L22

**Comment by enxebre** [comment]
fixes https://github.com/openshift/cluster-api-provider-aws/issues/119 cc @wking

File: pkg/actuators/machine/filter.go [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+)
+
+const (
+	clusterFilterKeyPrefix = "kubernetes.io/cluster/"
+	clusterFilterValue     = "owned"
+)
+
+func awsFilterName(name string) *string {
```
Comment (bison) [inline pkg/actuators/machine/filter.go]: Maybe this should be called `awsTagFilter` or something? To make it clear it's for filtering based on the given tag name.

Comment (spangenberg) [reply pkg/actuators/machine/filter.go]: :+1:

**Comment by wking** [comment]
/retest

CI was having issues earlier.

**Comment by enxebre** [comment]
@spangenberg can you please re-vendor to update latest https://github.com/openshift/cluster-api-actuator-pkg/pull/63

**Comment by enxebre** [comment]
needs https://github.com/openshift/cluster-api-provider-aws/pull/196

**Comment by wking** [comment]
[e2e-aws-operator][1]:

```
Summarizing 3 Failures:

[Fail] [Feature:MachineHealthCheck] MachineHealthCheck controller [It] should delete unhealthy machine 
/go/src/sigs.k8s.io/cluster-api-provider-aws/vendor/github.com/openshift/cluster-api-actuator-pkg/pkg/e2e/machinehealthcheck/machinehealthcheck.go:52

[Fail] [Feature:Operators] Machine API operator deployment should [It] reconcile controllers deployment 
/go/src/sigs.k8s.io/cluster-api-provider-aws/vendor/github.com/openshift/cluster-api-actuator-pkg/pkg/e2e/operators/machine-api-operator.go:28

[Fail] [Feature:Machines] Managed cluster should [It] recover from deleted worker machines 
/go/src/sigs.k8s.io/cluster-api-provider-aws/vendor/github.com/openshift/cluster-api-actuator-pkg/pkg/e2e/infra/infra.go:224
```

[actuator-pkg-staleness][2] is also suggesting changes, but I dunno if we need those to land or not?

[1]: https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_cluster-api-provider-aws/192/pull-ci-openshift-cluster-api-provider-aws-master-e2e-aws-operator/488
[2]: https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_cluster-api-provider-aws/192/pull-ci-openshift-cluster-api-provider-aws-master-actuator-pkg-staleness/187

**Pull request merged**